data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/SqsWorker.java

```diff
@@ -228,14 +232,34 @@ && isEventBridgeEventTypeCreated(parsedMessage)) {
             AcknowledgementSet acknowledgementSet = null;
             if (endToEndAcknowledgementsEnabled) {
                 // Acknowledgement Set timeout is slightly smaller than the visibility timeout;
-                int timeout = (int) sqsOptions.getVisibilityTimeout().getSeconds() - 2;
-                acknowledgementSet = acknowledgementSetManager.create((result) -> {
-                    acknowledgementSetCallbackCounter.increment();
-                    // Delete only if this is positive acknowledgement
-                    if (result == true) {
-                        deleteSqsMessages(waitingForAcknowledgements);
-                    }
-                }, Duration.ofSeconds(timeout));
+                int visibilityTimeout = (int)sqsOptions.getVisibilityTimeout().getSeconds();
+                System.out.println("=====++VISIBILITY TIMEOUT==="+visibilityTimeout+"...."+parsedMessage);
+                int expiryTimeout = 10*visibilityTimeout;
+                int progressCheckInterval = visibilityTimeout/2 - 1;
+                acknowledgementSet = acknowledgementSetManager.create(
+                    (ratio) -> {
+                        final int newVisibilityTimeoutSeconds = (int) (visibilityTimeout);
+                        final ChangeMessageVisibilityRequest changeMessageVisibilityRequest = ChangeMessageVisibilityRequest.builder()
```

Comment:
I tend to think that we should make this a configuration option. It is changing existing behavior and is now requiring access to a new API and thus requires new permissions.

Also, we should let the user configure a maximum. What if the acknowledgement set is stuck for some reason and it never completes after some number of hours?